lib/cocoapods-core/dependency.rb

```diff
@@ -109,11 +109,16 @@ def initialize(name = nil, *requirements)
     #         one or more version restrictions).
     #
     def requirement
-      if specific_version
-        Requirement.new(Version.new(specific_version.version))
-      else
-        @requirement
-      end
+      @specific_requirement || @requirement
```

Comment:
`requirement` is accessed a large number of times during the dependency analysis phase. for cases when `specific_version` is non-nil, this becomes an expensive codepath